0-bootstrap/variables.tf

```diff
@@ -91,77 +81,57 @@ variable "bucket_tfstate_kms_force_destroy" {
 /* ----------------------------------------
     Specific to Groups creation
    ---------------------------------------- */
-variable "groups" {
+variable "required_groups" {
   description = "Contain the details of the Groups to be created."
   type = object({
-    create_groups   = bool
-    billing_project = string
-    required_groups = object({
+    create_groups   = optional(bool, false)
+    billing_project = optional(string, null)
+    groups = object({
       group_org_admins           = string
       group_billing_admins       = string
-      billing_data_users         = string
-      audit_data_users           = string
-      monitoring_workspace_users = string
+      billing_data_users         = optional(string, "")
```

Comment:
Hello @eeaton 
I made this change on the code using the second approach!

Now the code have only one object with the two groups: `required_groups` and `optional_groups`.
Also have two different fields to toggle the creation of `required` or `optional` groups independently.
- If one of those two fields are **true** the user must provide a `billing_project`.

Reply:
Hello @eeaton 
I was wrong on that change. The groups _billing_data_users_, _audit_data_users_, _monitoring_workspace_users_ are also required but in other steps (_1-org_ and _2-environments_)

To simplify the code, we think in this approach:
1. Collapse the two variables into one **groups** variable, with `required_groups` and `optional_groups`. (like you mention in your second approach)
2. Two different flags to create the groups: `create_required_groups` and `create_optional_groups`. Both optional with the default **false**
3. All groups inside `optional_groups` are **optional** and when `create_optional_groups` is **true** will only create the the groups that was provided.
4. Inside `required_groups` we will have the required groups for all steps: **group_org_admins**, **group_billing_admins**, **billing_data_users**, **audit_data_users** and **monitoring_workspace_users**.
5. The step _0-bootstrap_ will use the values provided on the variable. And will output this values.
6. The other steps (_1-org_ and _2-environments_) will get the values from the _remote state_ (the output of _0-bootstrap_)


**What do you think of this approach?**

Reply:
That approach makes sense, thanks.

Tangent: It makes sense that at minimum Billing Admin and Org Admin are required, but it seemed odd to me that `monitoring_workspace_users` is required but other operational roles like `gcp_security_reviewer` are not. I expected that the infra should still deploy with or without the group.  I see it's because TF code explicitly creates the [IAM role for the monitoring group](https://github.com/terraform-google-modules/terraform-example-foundation/blob/dd1952d171cd369f2de534f28ed678dca71acb12/2-environments/modules/env_baseline/iam.tf#L21), but it [does not create any IAM role for the other optional groups](https://github.com/search?q=repo%3Aterraform-google-modules%2Fterraform-example-foundation%20gcp_security_reviewer&type=code). 

There's still an outstanding action item to redesign the IAM permissions for each group, so I'll create an issue to assign IAM roles for every group in the next version. Not a blocker for this PR.